### DISCUSSION TITLE
Potential error with MapStruct compiler argument

### DISCUSSION THREAD

**Discussion by Sax388** [spring-projects/spring-batch#4596] [Q&A] [answered]
When adding the compiler flag to use spring injection for MapStruct it seems there is a problem with Spring Batch. I haven't been able to provide a minimal project to reproduce it yet, the issue is detailed here: https://github.com/mapstruct/mapstruct/issues/3608.

Is there any change that the error is on the Spring Batch side? I've got two Spring projects, one works fine with the compiler flag, but the Spring Batch one goes crazy (see errors in issue above).

**Comment by fmbenhassine** [comment, marked as answer]
Apologies for the late reply, I was in a long leave. Have you fixed your issue since then?

I don't think this is related to Spring Batch, but if it is the case, I am willing to introduce the required changes in Spring Batch to make things work seamlessly with MapStruct.

**Reply by Sax388** [reply]
I currently don't have time to provide an MRE and I actually follow your assumption. Thanks for coming back to me, I'm always delighted from seeing you answering the questions of the community so thoughtfully.

**Reply by fmbenhassine** [reply]
Thank you! Always happy to help.
